profiles/lib/profile_debugger.lua

```diff
@@ -0,0 +1,138 @@
+-- Enable calling our lua profile code directly from the lua command line,
+-- which makes it easier to debug.
+-- We simulate the normal C++ environment by defining the required globals and functions.
+
+-- See debug_example.lua for an example of how to require and use this file.
+
+-- for more convenient printing of tables
+local pprint = require('lib/pprint')
+
+
+-- globals that are normally set from C++
+
+-- profiles code modifies this table
+properties = {}
+
+-- should match values defined in include/extractor/guidance/road_classification.hpp
+road_priority_class = {
+  motorway = 0,
+  trunk = 2,
+  primary = 4,
+  secondary = 6,
+  tertiary = 8,
+  main_residential = 10,
+  side_residential = 11,
+  link_road = 14,
+  bike_path = 16,
+  foot_path = 18,
+  connectivity = 31,
+}
+
+-- should match values defined in include/extractor/travel_mode.hpp
+mode = {
+  inaccessible = 0,
+  driving = 1,
+  cycling = 2,
+  walking = 3,
+  ferry = 4,
+  train = 5,
+  pushing_bike = 6,
+}
+
+-- Mock C++ helper functions which are called from LUA.
+-- TODO
+-- Debugging LUA code that uses these will not work correctly
+-- unless we reimplement the methods in LUA.
+
+function durationIsValid(str)
+  return true
+end
+
+function parseDuration(str)
+  return 1
+end
+
+function canonicalizeStringList(str)
+  return str
+end
```

Comment:
I'm only now seeing it but this means every time we bind a C++ function to Lua we must not forget to add a stub here too..